notebooks/a01_manifold_class.ipynb

```diff
@@ -0,0 +1,1630 @@
+{
```

Comment:
<p><code>dim</code> and <code>default_point_type</code> are actually "attributes" (they are just variables storing values), while <code>belongs</code>, <code>is_tangent</code>,<code>random_point</code> are "methods" (they are functions, that return an output when given an input).</p><p> </p><p>Remove: "(we will discuss tangent spaces in future notebooks)." + (this will make more sense after we discuss tangent spaces in future notebooks).</p><p> </p><p>We can remove "default_point_type" since it is not commented on.</p><p> </p><p>I would link it a bit more with the code here. With something like: "While the abstract methods in Manifold do not contain any implementation, the methods of the subclasses of Manifold, such as the Hypersphere subclass, are implemented and can be run."</p><p> </p><p>Attributes:</p><p> </p><ul><li><code>dim</code>  : For example, if we build an hypersphere of dimension 2 with the following code, we can check that sphere.dim gives back 2.</li></ul><p> </p><pre class="ql-syntax" spellcheck="false">from geomstats.geometry.hypersphere import Hypersphere

sphere = Hypersphere(dim=2)
print(f"The dimension of the sphere is {sphere.dim}")
</pre><p> </p><p>Methods:</p><ul><li><code>belongs</code> : For example, the point (0, 0, 1) belongs to the sphere we just built (it is the "north pole" of the sphere):</li></ul><pre class="ql-syntax" spellcheck="false">import geomstats.backend as gs

sphere.belongs(gs.array([0, 0, 1]))
</pre><ul><li><code>is_tangent</code> : for example, the vector (1, 1, 0 ) is tangent to the sphere at the north pole, since it does not have a vertical component (last component is equal to 0) </li></ul><pre class="ql-syntax" spellcheck="false">sphere.is_tangent(
  vector=gs.array([1, 1, 0]), 
  base_point=gs.array([0, 0, 1]))
</pre><p> </p><ul><li><code>random_point</code> : try generating a point and verify that the point belongs to the sphere? </li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/1340/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Reply:
<p>done</p>